### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by amunra**
Initial implementation, see `README.md` for details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,59 @@
...
+
+```plain
+service::key1=value1;key2=value2;key3=value3;
+```
+
+A few rules:
+* The last semicolon is mandatory.
+* Service name and keys are case-insensitive.
+* Values are case-sensitive.
+* A semicolon can't appear as a key.
```
Comment (jerrinot) [inline README.md]: nit: it should be _"in a key"_?

Comment (amunra) [reply README.md]: In actual fact, I've restricted it further since then: Keys are now alphanumeric, starting with a letter.
I'll tweak it.

**Review by jerrinot — APPROVED** [review]
both the impl and the spec look very nice!

one possible improvement: document FFI. things like: "how to test it". this is what I did:
```shell
cmake .
make
```
This generated a binary `test` which indeed runs the test. but I can see there is also a `run` script, which is trying to execute `./build/test`. But my binary is in the working dir, not inside `build`. 

this is no biggie, but documenting the blessed steps how to run the test would be nice. 

(after writing all of the above, I noticed there is the `compile` script shows everything. and this script is used on CI. so I guess a sentence in FFI readme will do)

**Review by jerrinot — CHANGES_REQUESTED** [review]
I realized it's actually overly restrictive - we do need underscore in keys. right now, the grammar does not make it possible. and indeed, this is failing:
```rust
#[test]
fn key_with_underscore() {
    let input = "https::addr=localhost;tls_roots=/etc/ssl/certs/ca-certificates.crt;";
    let config = parse_conf_str(input);
    assert!(!config.is_err());
}
```

File: questdb-confstr/tests/tests.rs [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+fn byte_not_char_position() {
+    let input = "p::n=静;:=42;";
+    let config = parse_conf_str(input);
+    assert!(config.is_err());
+    let err = config.unwrap_err();
+    assert_eq!(err.kind(), ErrorKind::ExpectedIdentifierNot(':'));
+    assert_eq!(err.position(), 9);
+    assert_eq!(
+        err.to_string(),
+        "expected identifier to start with ascii letter, not ':' at position 9"
```
Comment (jerrinot) [inline questdb-confstr/tests/tests.rs:301]: it looks like it's counting bytes and not chars. is that intentional? If I look at the input string then I don't see `:` at position 9.

Comment (amunra) [reply questdb-confstr/tests/tests.rs:301]: Yes, this _is_ intentional (and there's tests for it). It's because conting chars isn't really a thing, while you can slice the string at that point with the byte offset.

Comment (amunra) [reply questdb-confstr/tests/tests.rs:301]: I could track both, but for the few times anyone is ever going to use unicode chars and tell the difference it feels like an overkill.

Comment (jerrinot) [reply questdb-confstr/tests/tests.rs:301]: that's what I thought. but as a user - I would find it misleading. I don't run UTF-8 encoder in my head to tell where exactly is the position 9:)

I agree it's an edge-case and not worth spending much effort on it.

Comment (amunra) [reply questdb-confstr/tests/tests.rs:301]: Edge case, I think it's above good enough as-is.

**Review by jerrinot — APPROVED** [review]
